trace/workflow_execution_update.go

```diff
@@ -0,0 +1,115 @@
+package trace
+
+import (
+	"context"
+	"fmt"
+	"github.com/alitto/pond"
```

Comment:
I'm not sure if there's a vanilla workerpool. I'm happy to implement it myself (I did this initially) but I believe it might be better in the long run to use a well-tested workerpool.

Reply:
I was thinking just use goroutines and a semaphore.